packages/@ember/-internals/glimmer/tests/integration/helpers/invoke-helper-test.js

```diff
@@ -0,0 +1,272 @@
+        let count = 0;
+
+        class PlusOneHelper extends Helper {
+          compute([num]) {
+            count++;
+            return num + 1;
+          }
+        }
+
+        class PlusOne {
+          @tracked number;
+
+          constructor(number) {
+            this.number = number;
+          }
+
+          plusOne = invokeHelper(this, PlusOneHelper, () => {
+            return { positional: [this.number] };
+          });
+
+          get value() {
+            return getValue(this.plusOne);
+          }
+        }
+
+        let instance = new PlusOne(4);
+
+        assert.equal(instance.value, 5, 'helper works');
+        assert.equal(instance.value, 5, 'helper works');
+        assert.equal(count, 1, 'helper only called once');
+
+        instance.number = 5;
+
+        assert.equal(instance.value, 6, 'helper works');
+        assert.equal(count, 2, 'helper called a second time');
+      }
+
+      ['@test helper updates based on internal state changes'](assert) {
+        let count = 0;
+        let helper;
+
+        class PlusOneHelper extends Helper {
+          @tracked number = 4;
+
+          constructor() {
+            super(...arguments);
+            helper = this;
+          }
+
+          compute() {
+            count++;
+            return this.number + 1;
+          }
+        }
+
+        class PlusOne {
+          plusOne = invokeHelper(this, PlusOneHelper);
+
+          get value() {
+            return getValue(this.plusOne);
+          }
+        }
+
+        let instance = new PlusOne();
+
+        assert.equal(instance.value, 5, 'helper works');
+        assert.equal(instance.value, 5, 'helper works');
+        assert.equal(count, 1, 'helper only called once');
+
+        helper.number = 5;
+
+        assert.equal(instance.value, 6, 'helper works');
+        assert.equal(count, 2, 'helper called a second time');
+      }
+
+      ['@test helper destroys correctly when context object is destroyed'](assert) {
+        assert.expect(3);
+
+        let context = {};
+
+        class TestHelper extends Helper {
+          willDestroy() {
+            assert.ok(true, 'helper destroyed');
+          }
+        }
+
+        let helper = invokeHelper(context, TestHelper);
+
+        runTask(() => destroy(context));
+
+        assert.ok(isDestroyed(context), 'context destroyed');
+        assert.ok(isDestroyed(helper), 'helper cache destroyed');
+      }
+
+      ['@test helper destroys correctly when helper cache is destroyed'](assert) {
+        assert.expect(3);
+
+        let context = {};
+
+        class TestHelper extends Helper {
+          willDestroy() {
+            assert.ok(true, 'helper destroyed');
+          }
+        }
+
+        let helper = invokeHelper(context, TestHelper);
+
+        runTask(() => destroy(helper));
+
+        assert.notOk(isDestroyed(context), 'context NOT destroyed');
+        assert.ok(isDestroyed(helper), 'helper cache destroyed');
+      }
```

Comment:
Same as above, can you ensure the ordering relationship of the helper's own destroyables and the contexts?